PMM-10711 Relese Notes PMM 2-31.0 (#868)
percona/pmm-doc: docs/release-notes/2.31.0.md

## Release Highlights

#### Grafana 9.1 in PMM 2.31.0

We have upgraded Grafana to version 9.1 in PMM 2.31.0. With this upgrade, you can get a simplified user interface, enhanced visualizations, and a default Alerting experience. For more information, see What’s new in Grafana.
 
The following features have been introduced with the upgrade to Grafana to 9.1:
 
- **Expandable navbar (main menu)**
 
    For an enhanced user experience and better accessibility, the main menu has been expanded with crucial elements such as starred dashboards, system, MongoDB, MySQL, PostgreSQL, ProxySQL, and HAProxy.
 
- **Prometheus Query Builder**

    A new UI query builder has been introduced, by which you can write Prometheus queries effortlessly. Prometheus queries can be complex to grasp, but with this Grafana release in PMM, it becomes easier. 

- **Improvements to dashboard search**
    
    The dashboards can now be searched by panel title as well as dashboard title. Also, the overall performance has been enhanced as querying and sorting have become much faster.

> - **Default unified alerting**
> 
>     Grafana Alerting is now on by default.
> 

Reviewer: ```suggestion
```

There will be a separate section about Alerting.
Alerting is not really a Grafana 9 scope, Native Alerting was at 2.30 already.

## Vacuum monitoring dashboard

!!! caution alert alert-warning "Important"
    These experimental dashboards are subject to change. It is recommended to use these dashboards for testing purposes only.

A new experimental dashboard has been introduced to overcome the difficulties related to vacuum by providing them with the right and timely insights.

> ## Podman GA

Reviewer: ```suggestion
## New official (GA) deployment method - Podman 
```

> We are excited to announce the General Availability (GA) of Podman support for Deploying PMM 2.31.0. We had introduced it in 2.29.0 as a preview feature, but now we are production ready with this feature.

Reviewer: Can we add here a link to our Documentation for this deployment method?

Author: Sure.

## New Alerting

Percona Alerting is a solution based on Grafana Alerting with the addition of simplicity of Templates usage.  
This will enable users to use Alerting without deep knowledge of metrics details and consume the best knowledge from Percona Experts who created Alerting Templates. 

### Database as a Service (DBaaS)

In our constant endeavor and focus on an enhanced user experience, in PMM 2.31.0, we have simplified the deployment and configuration of DBaaS as follows:

- With PMM 2.31.0, you can easily add a DB cluster from the newly created K8s cluster. All the DB cluster window fields are auto-populated with the values based on the existing K8s cluster. 

- For PMM 2.31.0, while accessing DbaaS, if you have an existing Kubernetes cluster configured for DBaaS, you will be automatically redirected to the DB Cluster page. Otherwise, you would be redirected to the Kubernetes Cluster page.

> - We have added support for the latest (?) released Percona Operator for MongoDB (link to operator releases).

Reviewer: Probably need to be removed because https://jira.percona.com/browse/PMM-9989 might not be released.  

@gen1us2k, please let @rasika-chivate know the final result.

Author: @gen1us2k Please reaffirm...

Reviewer 2: The issue is still in progress. We do not support the 1.12 PSMDB operator yet.

Author: Will remove it from the RN..

Author: Done

> ### Operators support

Reviewer: ```suggestion
```

No need for a separate section. if this will be - this is a part of DBaaS highlights

Author: ok

Author: Done.

> # Percona Monitoring and Management 2.31.0

Reviewer 3: @rasika-chivate please also add changes for the left menu, RNs aren't visible there

Author: Yes. I will make the changes.

| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

!!! caution alert alert-warning "Important"
    We recommend using the latest version of PMM. This ensures that you have access to the latest PMM features and that your environment runs on the latest version of the underlying components, such as VictoriaMetrics, with all the bug fixes in place.

## Release Highlights

### Enhaced query building, navigation and Alerting

We have upgraded Grafana to version 9.1 in PMM 2.31.0. With this upgrade, you can get a simplified user interface, enhanced visualizations, and a default unified Alerting experience. For more information, see What’s new in Grafana.
 
The following features have been introduced with PMM 2.31.0:
 
- **Expandable navbar (main menu)**
 

>     For an enhanced user experience and better accessibility, the main menu has been expanded with crucial elements such as starred dashboards, system, MongoDB, MySQL, PostgreSQL, ProxySQL, and HAProxy.

Reviewer 4: I suggest introducing "Starred dashboards" as a separate item of the Expandable navbar.

Author: Sure

Author: Done

>     For an enhanced user experience and better accessibility, the main menu has been expanded with crucial elements such as starred dashboards, system, MongoDB, MySQL, PostgreSQL, ProxySQL, and HAProxy.

Reviewer 4: what about this:
.... the main menu has been designed by moving the dashboard navigation from the PMM Dashboards section to the main menu for quick access. So you can now browse dashboards from the navbar like Operating System (OS), MySQL, MongoDB, PostgreSQL, ProxySQL, and HAProxy. And also new menu provides an expandability function to see all sections on the menu.

 
- **Prometheus Query Builder**

    A new UI query builder has been introduced, by which you can write Prometheus queries effortlessly. Prometheus queries can be complex to grasp, but with this Grafana release in PMM, it becomes easier. 

- **Improvements to dashboard search**
    
    The dashboards can now be searched by panel title as well as dashboard title. Also, the overall performance has been enhanced as querying and sorting have become much faster.

> 

Reviewer 4: Can we also mention Explore to dashboard feature that was introduced in Grafana 9: https://grafana.com/docs/grafana/latest/whatsnew/whats-new-in-v9-0/#explore-to-dashboard

Author: Done

> 

Reviewer 4: can we add 1-line info for : https://grafana.com/docs/grafana/latest/whatsnew/whats-new-in-v9-0/#command-palette

Author: Sure.

A new experimental dashboard has been introduced to overcome the difficulties related to vacuum by providing them with the correct and timely insights.

This dashboard contains the following:

- Dead tuples - Identifies the number of dead rows in each table even though the rows are physically removed from the table.

- Last time vacuum ran - Tracks the last time a vacuum or autovacuum process successfully ran on each of your tables.

- Number of rows modified since last analyze - The number of rows changed since the last time ANALYZE ran.

- Manual vacuum events - Tracks the number of times a manual vacuum was run on each table. 

- Table disk usage - Tracking the disk space used by each table is crucial as it enables you to gauge expected changes in the query performance over time—but it can also help you detect potential vacuuming-related issues.

## New official (GA) deployment method - Podman 

We are excited to announce the General Availability (GA) of Podman support for Deploying PMM 2.31.0. We had introduced it in 2.29.0 as a preview feature, but now we are production ready with this feature.

## New Alerting

> Percona Alerting is a solution based on Grafana Alerting with the addition of simplicity of Templates usage.  

Reviewer 5: ```suggestion
Percona Alerting is a solution based on Grafana Alerting with the added simplicity of Percona's Alerting Templates.
```

This will enable users to use Alerting without deep knowledge of metrics details and consume the best knowledge from Percona Experts who created Alerting Templates. 

### Simplied deployment with Database as a Service (DBaaS)

In our constant endeavor and focus on an enhanced user experience, in PMM 2.31.0, we have simplified the deployment and configuration of DBaaS as follows:

- With PMM 2.31.0, you can easily add a DB cluster from the newly created K8s cluster. All the DB cluster window fields are auto-populated with the values based on the existing K8s cluster. 

- For PMM 2.31.0, while accessing DbaaS, if you have an existing Kubernetes cluster configured for DBaaS, you will be automatically redirected to the DB Cluster page. Otherwise, you would be redirected to the Kubernetes Cluster page.

> - PMM now supports Percona Operator for MongoDB [1.12.0](https://docs.percona.com/percona-operator-for-mongodb/RN/Kubernetes-Operator-for-PSMONGODB-RN1.12.0.html).

Reviewer 3: looks like we don't ? CC @gen1us2k @rnovikovP

Author: Removed from the RN.

## New Features

- [PMM-10089](https://jira.percona.com/browse/PMM-10089): We have upgraded Grafana to 9.1 in PMM 2.31.0.

- [PMM-10092](https://jira.percona.com/browse/PMM-10092): We have expanded and improved the main menu for PMM 2.31.0 based on Grafana 9.1. The main menu contains crucial elements, one of them being starred dashboards. 

> - [PMM-10467](https://jira.percona.com/browse/PMM-10467):  

Reviewer 3: @rasika-chivate @artemgavrilov no descr  here

Reviewer 6: Because there is no ready implementation yet. I will provide more info later.

Author: @artemgavrilov @catalinaadam Could you please provide the content for this.

> - [PMM-10443](https://jira.percona.com/browse/PMM-10443): 

Reviewer 3: @rasika-chivate @JiriCtvrtka no desc here

Author: @JiriCtvrtka @BupycHuk Please provide your inputs here.

Reviewer 7: @rasika-chivate Agree with Denys, no desc needed.

Reviewer 8: As far as I see it was highlighted above and in the next item.

> - [PMM-10244](https://jira.percona.com/browse/PMM-10244): 

Reviewer 3: @rasika-chivate this is Vacuum feature, covered in 10443.

Author: No @denisok Need inputs from @BupycHuk or @JiriCtvrtka

Reviewer 2: It's part of Vacuum feature covered in 10443

Author: ok

- [PMM-10064](https://jira.percona.com/browse/PMM-10064): Introduced UI changes on the PMM Inventory pages for better readability and user experience.

- [PMM-10076](https://jira.percona.com/browse/PMM-10076): For an improved UI experience, we have enforced the enter button for the date picker, which performs the same action as clicking `apply time range`.

- [PMM-7806](https://jira.percona.com/browse/PMM-7806): Upgraded postgres_exporter version used in pmm from 0.8.0 to 0.10.1. With this upgrade, you can access the latest features with all the bug fixes in place. 

  
- [PMM-10018](https://jira.percona.com/browse/PMM-10018): Lowered the default values of CPU and memory for haproxy while deploying a DBaaS [Percona XtraDB Cluster](https://www.percona.com/software/mysql-database/percona-xtradb-cluster) (PXC) cluster to avoid wasting resources.

    
- [PMM-9118](https://jira.percona.com/browse/PMM-9118): Refined the error message thrown when a secure connection cannot be established while adding a monitoring database to PMM, thus making it much more contextual and easy to troubleshoot.

> - [PMM-10516](https://jira.percona.com/browse/PMM-10516): 

Reviewer 5: ```suggestion
- [PMM-10516](https://jira.percona.com/browse/PMM-10516): Added support for MongoDB physical backups. With this, PMM backup management now supports both logical and physical backups for MongoDB services that are based on [Percona Server for MongoDB](https://www.percona.com/software/mongodb/percona-server-for-mongodb)
```

Reviewer 5: Also, I wonder if this counts as an improvement or new feature

- [PMM-10095](https://jira.percona.com/browse/PMM-10095): While removing the service with pmm-admin, now you do not have to specify the service name for a single service. PMM will automatically look up this service and remove it.

> - [PMM-9989](https://jira.percona.com/browse/PMM-9989): PMM now supports Percona Operator for MongoDB 1.12.0.

Reviewer 3: no, we don't. CC @gen1us2k @puneet0191 ticket is still In Progress

Reviewer 2: The issue is still in progress. We do not support the 1.12 PSMDB operator yet.

Author: Will remove it..

- [PMM-10587](https://jira.percona.com/browse/PMM-10587): Fixed an issue where if you hover over the telemetry icon stacktrace with error appears.

- [PMM-10513](https://jira.percona.com/browse/PMM-10513): Fixed an issue where pmm-admin summary stored PMM password in client/status.json, thus posing a security risk.

- [PMM-10065](https://jira.percona.com/browse/PMM-10065): Fixed an issue encountered while monitoring a large number of nodes (1000) and services. Here, the changing scrape frequency led to data not being captured on the dashboard and throwing a templating error.

- [PMM-9973](https://jira.percona.com/browse/PMM-9973): Fixed an issue where the CPU utilization for Postgres Exporter was on the higher side.

- [PMM-9891 ()]https://jira.percona.com/browse/PMM-9891): Fixed an issue where QAN truncated long queries.

> - [PMM-9556](https://jira.percona.com/browse/PMM-9556): Fixed an issue where creating a database cluster for MongoDB 5.0x was failing in DBaaS.

Reviewer 3: This wasn't fixed. @percona-csalguero we probably need to remove this and add Alert/Note that ppl shouldn't use Mongo 5 with operators 1.10, 1.11 ?

Author: ok

## Release Highlights

### Enhaced query building, navigation and Alerting

We have upgraded Grafana to version 9.1 in PMM 2.31.0. With this upgrade, you can get a simplified user interface, enhanced visualizations, and a default unified Alerting experience. For more information, see What’s new in Grafana.
 
The following features have been introduced with PMM 2.31.0:
 
- **Expandable navbar (main menu)**
 
    For an enhanced user experience, we have expanded and designed the main menu d by moving the dashboard navigation from the PMM Dashboards menu to the main menu for quick access. So you can now browse dashboards like Operating System (OS), MySQL, MongoDB, PostgreSQL, ProxySQL, and HAProxy directly from the navbar.  

- **Prometheus Query Builder**

    We have introduced a new UI query builder using which you can write Prometheus queries effortlessly. Prometheus queries can be complex to grasp, but with this Grafana release in PMM, it becomes easier. 

- **Starred dashboards**

>    To enhance user experience further, we have introduced Starred dashboards, where you can access your favorite dashboards directly from the main menu. 

Reviewer 3: it still doesn't render good. Looks shifted

## Vacuum monitoring dashboard

!!! caution alert alert-warning "Important"
    This experimental dashboard is subject to change. It is recommended to use this dashboard for testing purposes only.

A new experimental dashboard was released to help our users gain timely insights into the autovacuum process in PostgreSQL. This dashboard is designed to help fine-tune the vacuum configuration and to prevent a XID wraparound - both of which directly contribute to improved performance of the database.

This dashboard contains the following:

- Dead tuples - Identifies the number of dead rows in each table even though the rows are physically removed from the table.

- Last time vacuum ran - Tracks the last time a vacuum or autovacuum process successfully ran on each of your tables.

- Number of rows modified since last Analyze - The number of rows changed since the last time ANALYZE ran.

- Manual vacuum events - Tracks the number of times a manual vacuum was run on each table. 

> - Table disk usage - Tracking the disk space used by each table is crucial as it enables you to gauge expected changes in the query performance over time—but it can also help you detect potential vacuuming-related issues.

Reviewer 9: ```suggestion
- Table disk usage - Tracking the disk space used by each table is crucial as it enables you to gauge expected changes in the query performance over time - but it can also help you detect potential vacuuming-related issues.
```

This will enable users to use Alerting without deep knowledge of metrics details and consume the best knowledge from Percona Experts who created Alerting Templates. 

### Simplied deployment with Database as a Service (DBaaS)

In our constant endeavor and focus on an enhanced user experience, in PMM 2.31.0, we have simplified the deployment and configuration of DBaaS as follows:

- With PMM 2.31.0, you can easily add a DB cluster from the newly created K8s cluster. All the DB cluster window fields are auto-populated with the values based on the existing K8s cluster. 

- For PMM 2.31.0, while accessing DbaaS, if you have an existing Kubernetes cluster configured for DBaaS, you will be automatically redirected to the DB Cluster page. Otherwise, you would be redirected to the Kubernetes Cluster page.

- We have added support for the latest (?) released Percona Operator for MongoDB (link to operator releases).

### Operators support

PMM now supports Percona Operator for MongoDB [1.12.0](https://docs.percona.com/percona-operator-for-mongodb/RN/Kubernetes-Operator-for-PSMONGODB-RN1.12.0.html).

## New Features

> - [PMM-10089](https://jira.percona.com/browse/PMM-10089): We have upgraded Grafana to 9.1 in PMM 2.31.0.

Reviewer 8: I think this item should be highlighted because we upgraded the major version of Grafana.

Author: @BupycHuk Its already a part of Release Highlights

- [PMM-10516](https://jira.percona.com/browse/PMM-10516): 

- [PMM-10095](https://jira.percona.com/browse/PMM-10095): While removing the service with pmm-admin, now you do not have to specify the service name for a single service. PMM will automatically look up this service and remove it.

- [PMM-9989](https://jira.percona.com/browse/PMM-9989): PMM now supports Percona Operator for MongoDB 1.12.0.

## Bugs Fixed

- [PMM-10628](https://jira.percona.com/browse/PMM-10628): While adding MongoDB or PostgreSQL database via socket, PMM was throwing an error. This has been fixed in PMM 2.31.0.

- [PMM-10624](https://jira.percona.com/browse/PMM-10624): Empty dataname label after PostgreSQL upgraded in PMM.

- [PMM-10608](https://jira.percona.com/browse/PMM-10608): Fixed issues pertaining to the new Home Dashboard, such as high CPU panel unreadable for a higher number of nodes (more than ten nodes), filtering for the environment as well as node name not working correctly, time range not working as expected, some panels on the dashboard not displaying data for selected filters, and so on.

> - [PMM-10587](https://jira.percona.com/browse/PMM-10587):Fixed an issue where if you hover over the telemetry icon stacktrace with error appears.

Reviewer 8: ```suggestion
- [PMM-10587](https://jira.percona.com/browse/PMM-10587): Fixed an issue where if you hover over the telemetry icon stacktrace with error appears.
```

- [PMM-10513](https://jira.percona.com/browse/PMM-10513): Fixed an issue where pmm-admin summary stored PMM password in client/status.json, thus posing a security risk.

> - [PMM-10065](https://jira.percona.com/browse/PMM-10065): Fixed an issue encountered while monitoring a large number of nodes (1000) and services. Here, the changing scrape frequency led to data not being captured on the dashboard and throwing a templating error.

Reviewer 8: @anton-bystrov have we fixed it completely or only in a new dashboard?

Reviewer 3: we have adjusted some VM defaults that would prevent it for some large number of monitoring services. But also probably still depends how many services you have :)

Original issue was fixed.

Author: @BupycHuk resolve this comment?

- [PMM-9973](https://jira.percona.com/browse/PMM-9973): Fixed an issue where the CPU utilization for Postgres Exporter was on the higher side.

> - [PMM-9891 ()]https://jira.percona.com/browse/PMM-9891): Fixed an issue where QAN truncated long queries.

Reviewer 8: This one is also won't be included

Author: Will remove it.

- [PMM-9556](https://jira.percona.com/browse/PMM-9556): Fixed an issue where creating a database cluster for MongoDB 5.0x was failing in DBaaS.

- [PMM-10044](https://jira.percona.com/browse/PMM-10044): Fixed an issue where QAN displayed enormously high values.

> - [PMM-10213](https://jira.percona.com/browse/PMM-10213): Fixed an issue where adding MySQL with the defaults-file parameter failed.

Reviewer 8: this one won't be included

Author: will remove it.

> - [PMM-10231](https://jira.percona.com/browse/PMM-10231): Fixed an issue for MySQL, where the --credential-source parameter ignored the username present in the json file.

Reviewer 8: this one won't be included

Author: ok. Will remove it.

The **Alert Rules** tab has also evolved into a more efficient interface with an added layer for simplifying complex Grafana rules. You’ll find that the new **Percona templated alert** option here offers the same functionality available in the old Integrated Alerting, but uses a friendlier interface, powered by Grafana’s advanced alerting capabilities. 

As an important and generally useful feature, this new Alerting feature is now enabled by default and ready to use in production! 

### Deprecated Integrated Alerting
The new Percona Alerting feature fully replaces the old Integrated Alerting in previous PMM versions. The new alerting brings full feature parity with Integrated Alerting, along with additional benefits like Grafana-based alert rules and a unified alerting command center. 

If you have been using Integrated Alerting in the previous PMM version, your custom alert rule templates will be automatically migrated to PMM 2.31. After upgrading to this new version, you will find all your alert templates under **Alerting > Alert Templates**. 
These templates are also available for creating new alert rules using the new Percona Templated Alerts option on the Alert Rules tab. 

> However, custom alert rules created with Integrated Alerting are not automatically migrated to Percona Alerting. After upgrading, make sure to manually migrate any custom alert rules that you want to transfer to PMM 2.31 using the script that will be provided soon. 

Reviewer 6: Just `alert rules`, because there is no other rules.
```suggestion
However, alert rules created with Integrated Alerting are not automatically migrated to Percona Alerting. After upgrading, make sure to manually migrate any custom alert rules that you want to transfer to PMM 2.31 using the script that will be provided soon. 
```

!!! caution alert alert-warning "Important"
    We recommend using the latest version of PMM. This ensures that you have access to the latest PMM features and that your environment runs on the latest version of the underlying components, such as VictoriaMetrics, with all the bug fixes in place.

## Release Highlights

### Enhaced query building, navigation and Alerting

We have upgraded Grafana to version 9.1 in PMM 2.31.0. With this upgrade, you can get a simplified user interface, enhanced visualizations, and a default unified Alerting experience. For more information, see What’s new in Grafana.
 
The following features have been introduced with PMM 2.31.0:
 
- **Expandable navbar (main menu)**
 
    For an enhanced user experience, we have expanded and designed the main menu d by moving the dashboard navigation from the PMM Dashboards menu to the main menu for quick access. So you can now browse dashboards like Operating System (OS), MySQL, MongoDB, PostgreSQL, ProxySQL, and HAProxy directly from the navbar.  

- **Prometheus Query Builder**

>    We have introduced a new UI query builder using which you can write Prometheus queries effortlessly. Prometheus queries can be complex to grasp, but with this Grafana release in PMM, it becomes easier. 

Reviewer 3: on render ident doesn't look right:
https://pmm-v2-31-0-pr-868.onrender.com/release-notes/2.31.0.html

- **Starred dashboards**

>    To enhance user experience further, we have introduced Starred dashboards, where you can access your favorite dashboards directly from the main menu. 

Reviewer 3: on render ident doesn't look right:
https://pmm-v2-31-0-pr-868.onrender.com/release-notes/2.31.0.html

Author: Fixed
